### PR TITLE
Feature gateway 3210 trusted sender

### PR DESCRIPTION
**Pull request by gugo1**
https://messagemedia.atlassian.net/browse/GATEWAY-3210
Add a new tag:
- name: Source Address
- description: The Sender Address API provides a number of endpoints for you to request a Sender ID and gain real-time access to its approval status
https://messagemedia.atlassian.net/wiki/spaces/PRD/pages/2741862739/Trusted+Sender+API+Reference 
- Request a sender address
- Get status of a sender address request
- Get all approved sender address

### CODE REVIEW

File: spec/openapi.yaml [hunk 1]
```diff
@@ -1415,6 +1418,172 @@ paths:
...
+                    last_modified_date: 2023-08-24T04:21:55.958Z
+                pagination:
+                  page_size: 20
+        '403':
+          description: Unauthorised
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/403response'
+        '404':
```
Comment (paultirta) [inline spec/openapi.yaml]: I don't think there's a 404 response for this endpoint

File: spec/openapi.yaml [hunk 2]
```diff
@@ -1415,6 +1418,172 @@ paths:
...
+            type: string
+      responses:
+        '200':
+          description: Get the status of Sender Address Request
+          headers: {}
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SenderAddressRequestItem'
+        '403':
```
Comment (paultirta) [inline spec/openapi.yaml:1522]: Need 401 Unauthorized when auth is invalid

File: spec/openapi.yaml [hunk 3]
```diff
@@ -1415,6 +1418,172 @@ paths:
...
+                    destination_countries:
+                      - AU
+                    reason: 'This is my reason 3'
+                    label: 'This is my label 3'
+                    account_id: my_account,
+                    created_date: 2023-08-24T04:21:55.958Z,
+                    last_modified_date: 2023-08-24T04:21:55.958Z
+                pagination:
+                  page_size: 20
+        '403':
```
Comment (paultirta) [inline spec/openapi.yaml]: Same here, need 401 Unauthorized when auth is invalid

Comment (gugo1) [reply spec/openapi.yaml]: Done.

File: spec/openapi.yaml (hunk 2 above)
Comment (gugo1) [reply spec/openapi.yaml:1522]: Done.

File: spec/openapi.yaml (hunk 1 above)
Comment (gugo1) [reply spec/openapi.yaml]: Done.   404 is removed.

File: spec/openapi.yaml [hunk 4]
```diff
@@ -1415,6 +1418,170 @@ paths:
...
+            type: string
+      responses:
+        '200':
+          description: Get the status of Sender Address Request
+          headers: {}
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SenderAddressRequestItem'
+        '401':
```
Comment (paultirta) [inline spec/openapi.yaml:1520]: Also need 400 Bad Request if the id in the path is not uuid

File: spec/openapi.yaml [hunk 5]
```diff
@@ -1415,6 +1418,170 @@ paths:
...
+                    destination_countries:
+                      - AU
+                    reason: 'This is my reason 3'
+                    label: 'This is my label 3'
+                    account_id: my_account,
+                    created_date: 2023-08-24T04:21:55.958Z,
+                    last_modified_date: 2023-08-24T04:21:55.958Z
+                pagination:
+                  page_size: 20
+        '401':
```
Comment (paultirta) [inline spec/openapi.yaml:1612]: Also need 400 Bad Request when the query string values are invalid

**Review by milldaulagala — APPROVED** [review]

File: spec/openapi.yaml (hunk 5 above)
Comment (gugo1) [reply spec/openapi.yaml:1612]: Done. Add 400.

File: spec/openapi.yaml (hunk 4 above)
Comment (gugo1) [reply spec/openapi.yaml:1520]: Done. Add 400.

**Review by paultirta — APPROVED** [review]

**Review by david-porter-messagemedia — APPROVED** [review]

**Pull request merged**